microcosm_logging/levels.py

```diff
@@ -0,0 +1,53 @@
+"""
+Logging levels
+
+"""
+from functools import total_ordering
+
+
+@total_ordering
+class ConditionalLoggingLevel(object):
+    """
+    A logging level that is conditional on a function.
+
+    Works by masquerading as an int.
```

Comment:
`logger.setLevel()` expects either an `int` or one of the reserved logging level strings that can be converted to an int (e.g. `DEBUG`).  The logging component only _uses_ the comparison functions of the level.

Also, the `@total_ordering` decorator generates the _missing_ comparison functions given `eq` and (any) one other; I chose `lt`, we get `le`, `gt`, `ge`, `ne` for free